### ISSUE TITLE
Hash Checker

### ISSUE THREAD

**Issue by shuvashish76** [TeamAmaze/AmazeFileUtilities#121]
Feature request for hash checker integration.
https://github.com/hash-checker/hash-checker is probably the best foss alternative.

**Comment by VishalNehra** [comment]
I don't see how we can integrate this feature in this app. It's not related to files. This utilities app is focused on dealing with files, saving memory and media players. 
I think we shouldn't clutter with features not relevant.

**Comment by shuvashish76** [comment]
> The utilities app is focused on dealing with files.

Sometimes file corruption happens while sending big size files/backups to another device/PC, this feature helps for that purpose.

**Comment by VishnuSanal** [comment]
> > The utilities app is focused on dealing with files.
> 
> Sometimes file corruption happens while sending big size files/backups to another device/PC, this feature helps for that purpose.

Agreed for the purpose; but the mentioned repository is not a library but an actual implementation of the concept.

In the case of AFU, we should implement it from scratch. As Vishal mentioned, this is not on the priorities in the roadmap for AFU's development. Keeping this issue open as it is open for contributions!

**Comment by shuvashish76** [comment]
> mentioned repository is not a library but an actual implementation

Mentioned that here because the method they used works for very big size files ([link](https://github.com/CodeDead/DeadHash-android/issues/29#issuecomment-1611450372))
Thanks 👍🏿

**Comment by VishalNehra** [comment]
> > The utilities app is focused on dealing with files.
> 
> Sometimes file corruption happens while sending big size files/backups to another device/PC, this feature helps for that purpose.

We already have a hashing function to identify duplicate files. We can introduce hashing at the end of transfer to validate the file is correctly sent. That's not a problem. However introducing multiple hashing functions, or allowing user to generate hashes isn't really relevant here.
